### PR TITLE
WASM: Added support for DOM's CustomEvents

### PR DESCRIPTION
**Pull request by carldebilly**
## Feature

You can now add event listener for HTML's `CustomEvent` and get a string payload.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tested code with current [supported SDKs](../README.md#supported)
- [X] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [X] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [X] Contains **NO** breaking changes
- [X] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/UIElement.wasm.cs [hunk 1]
```diff
@@ -265,6 +265,8 @@ internal enum HtmlEventExtractor
 			TappedEventExtractor,
 			KeyboardEventExtractor,
 			FocusEventExtractor,
+			CustomEventDetailStringExtractor, // For use with CustomEvent("name", {detail:{string detail here}})
+			CustomEventDetailJsonExtractor, // For use with CustomEvent("name", {detail:{detail here}}) - will be STRIGNIFIED (json)
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/UIElement.wasm.cs]:
```suggestion
			CustomEventDetailJsonExtractor, // For use with CustomEvent("name", {detail:{detail here}}) - will be string-ified (json)
```

File: src/Uno.UI/UI/Xaml/UIElementExtensions.Wasm.cs [hunk 2]
```diff
@@ -19,5 +19,45 @@ public static void UnregisterHtmlEventHandler(this UIElement element, string eve
 		{
 			element.UnregisterEventHandler(eventName, handler);
 		}
+
+		public class HtmlCustomEventArgs : EventArgs
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/UIElementExtensions.Wasm.cs]: Other file

File: src/Uno.UI/UI/Xaml/UIElementExtensions.Wasm.cs [hunk 3]
```diff
@@ -19,5 +19,45 @@ public static void UnregisterHtmlEventHandler(this UIElement element, string eve
...
+		}
+
+		/// <summary>
+		/// Will invoke `addEventListener(<paramref name="eventName"/>)` on the corresponding HTML element.
+		/// </summary>
+		/// <remarks>
+		/// For use with CustomEvent("name", {detail:{detail here}}).
+		/// </remarks>
+		/// <param name="isDetailJson">
+		/// True will JSON.stringnify the content. False will treat the content as a string.
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/UIElementExtensions.Wasm.cs]:
```suggestion
		/// True will JSON.stringify the content. False will treat the content as a string.
```

File: src/Uno.UI/UI/Xaml/UIElementExtensions.Wasm.cs (hunk 2 above)
Comment (carldebilly) [reply src/Uno.UI/UI/Xaml/UIElementExtensions.Wasm.cs]: Done

**Pull request merged**